colossalai/zero/sharded_optim/sharded_optim.py

```diff
@@ -206,10 +214,10 @@ def _initialize_optimizer_states(self):
         # set this dummpy zero tensor as grad
         for group_id in range(len(self._fp32_flat_param_groups_of_current_rank)):
             fp32_partition_param = self._fp32_flat_param_groups_of_current_rank[group_id]
-            fp32_partition_grad = torch.zeros_like(fp32_partition_param)
+            fp32_partition_grad = torch.zeros_like(fp32_partition_param, dtype=fp32_partition_param.dtype)
```

Comment:
I think `torch.zero_like()` returns a tensor with the same dtype as the original tensor.

Reply:
yes, you are right